api/enki.py

```diff
@@ -539,226 +479,253 @@ def queue_response(self, status_code, headers={}, content=None):
-            identifier, is_new = metadata.primary_identifier.load(self._db)
-            if not identifier in identifiers:
-                # Enki told us about a book we didn't ask
-                # for. This shouldn't happen, but if it does we should
-                # do nothing further.
-                continue
-            seen_identifiers.add(identifier.identifier)
-            result = self.set_metadata(identifier, metadata)
-            if not isinstance(result, CoverageFailure):
-                result = self.handle_success(identifier)
-            batch_results.append(result)
-
-        # Create a CoverageFailure object for each original identifier
-        # not mentioned in the results.
-        for identifier_string in identifier_strings:
-            if identifier_string not in seen_identifiers:
-                identifier, ignore = Identifier.for_foreign_id(
-                    self._db, api.ENKI_ID, identifier_string
-                )
-                result = CoverageFailure(
-                    identifier, "Book not found in Enki", data_source=self.output_source, transient=True
-                )
-                batch_results.append(result)
-        return batch_results
-
-    def process_item(self, identifier):
-        results = self.process_batch([identifier])
-        return results[0]
-
-    def handle_success(self, identifier):
-        return self.set_presentation_ready(identifier)
+    log = logging.getLogger("Enki Bibliographic Parser")
 
-class BibliographicParser(object):
+    # Convert the English names of languages given in the Enki API to
+    # the codes we use internally.
+    LANGUAGE_CODES = {
+        "English": u"eng",
+        "French" : u"fre",
+        "Spanish": u"spa",
+    }
 
-    """Helper function to parse JSON"""
     def process_all(self, json_data):
-        data = json.loads(json_data)
-        returned_titles = data["result"]["titles"]
-	titles = returned_titles
+        if isinstance(json_data, basestring):
+            json_data = json.loads(json_data)
+        returned_titles = json_data.get("result", {}).get("titles", [])
 	for book in returned_titles:
-	    data = self.process_one(book)
+            data = self.extract_bibliographic(book)
             if data:
                 yield data
-    log = logging.getLogger("Enki Bibliographic Parser")
-    log.setLevel(logging.DEBUG)
-
-    def __init__(self, include_availability=True, include_bibliographic=True):
-        self.include_availability = include_availability
-        self.include_bibliographic = include_bibliographic
 
     def extract_bibliographic(self, element):
+        """Extract Metadata and CirculationData from a dictionary
+        of information from Enki.
+
+        :return: A Metadata with attached CirculationData.
+        """
+        # TODO: it's not clear what these are or whether we'd find them
+        # useful:
+        #  dateSaved
+        #  length
+        #  publishDate
```

Comment:
publishDate seems like it would be useful.

Reply:
The problem is I've never seen a value specified for publishDate, and since "publication date" can mean so many different things I can't tell where it should go if it were to show up.